cmd/thanos/sidecar.go

```diff
@@ -262,43 +262,52 @@ func runSidecar(
 
 		exemplarSrv := exemplars.NewPrometheus(conf.prometheus.url, c, m.Labels)
 
-		infoSrv := info.NewInfoServer(
-			component.Sidecar.String(),
-			info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
+		var infoOptions []info.ServerOptionFunc
+		if !promAgentModeEnabled {
+			infoOptions = append(infoOptions, info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
 				return promStore.LabelSet()
 			}),
-			info.WithStoreInfoFunc(func() *infopb.StoreInfo {
-				if httpProbe.IsReady() {
-					mint, maxt := promStore.Timestamps()
-					return &infopb.StoreInfo{
-						MinTime:                      mint,
-						MaxTime:                      maxt,
-						SupportsSharding:             true,
-						SupportsWithoutReplicaLabels: true,
-						TsdbInfos:                    promStore.TSDBInfos(),
+				info.WithStoreInfoFunc(func() *infopb.StoreInfo {
+					if httpProbe.IsReady() {
+						mint, maxt := promStore.Timestamps()
+						return &infopb.StoreInfo{
+							MinTime:                      mint,
+							MaxTime:                      maxt,
+							SupportsSharding:             true,
+							SupportsWithoutReplicaLabels: true,
+							TsdbInfos:                    promStore.TSDBInfos(),
+						}
 					}
-				}
-				return nil
-			}),
-			info.WithExemplarsInfoFunc(),
+					return nil
+				}),
+				info.WithExemplarsInfoFunc())
+		}
+		infoOptions = append(infoOptions,
 			info.WithRulesInfoFunc(),
 			info.WithTargetsInfoFunc(),
 			info.WithMetricMetadataInfoFunc(),
 		)
+		infoSrv := info.NewInfoServer(
+			component.Sidecar.String(), infoOptions...,
+		)
 
-		storeServer := store.NewLimitedStoreServer(store.NewInstrumentedStoreServer(reg, promStore), reg, conf.storeRateLimits)
-		s := grpcserver.New(logger, reg, tracer, grpcLogOpts, tagOpts, comp, grpcProbe,
-			grpcserver.WithServer(store.RegisterStoreServer(storeServer, logger)),
-			grpcserver.WithServer(rules.RegisterRulesServer(rules.NewPrometheus(conf.prometheus.url, c, m.Labels))),
+		opts := []grpcserver.Option{
 			grpcserver.WithServer(targets.RegisterTargetsServer(targets.NewPrometheus(conf.prometheus.url, c, m.Labels))),
+			grpcserver.WithServer(rules.RegisterRulesServer(rules.NewPrometheus(conf.prometheus.url, c, m.Labels))),
```

Comment:
Just curious, does Prometheus agent support rules now? 
I thought it is only available in normal Prometheus implementation.

Reply:
I see that alerting is disabled but I am not sure about evaluating rules.

Reply:
I think rule evaluation is also disabled. Prometheus agent doesn't have a query engine with it so no evaluation